test/server/gateways/postgres/plugin.test.ts

```diff
@@ -46,24 +50,53 @@ describe("Postgres Gateway Tests", function () {
       expect(
         pg.Client.prototype.query
       ).toHaveBeenCalledWith(
-        `INSERT INTO trades VALUES(${trade.id}, ${trade.ticker}, ${trade.company_name}, ${trade.reference_number}, ${trade.unit_price}, ${trade.quantity}, ${trade.total_cost}, ${trade.trade_type}, ${trade.note}, ${trade.created_at}, ${trade.trade_status}) RETURNING id`,
+        `INSERT INTO trades (id, ticker, company_name, reference_number, unit_price, quantity, total_cost, trade_type, note, created_at, trade_status) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING id`,
         ["abc", "ABC", "", "", 0, 0, 0, "long", "", now, true]
       );
     });
+
+    it("should throw an error if the query fails", async function () {
+      try {
+        pg.Client.prototype.query = jest
+          .fn()
+          .mockRejectedValueOnce(new TypeError("Fake Error"));
+        await db.write(trade);
+      } catch (error) {
+        expect(error.message).toBe("Fake Error");
```

Comment:
**Comment**: This is a poorly constructed test. What this unit test is saying is 'expect stubs to return the correct result', which is exactly what stubs are meant to do. This behavior does not need testing. Instead, the behavior we should be testing if `logger.log` has been successfully called or not.

Reply:
Updated all tests to spy on logger. Tests will assert logger has been called when an error occurs.